# Evaluating gene signatures in single-cell datasets  
  
List of computing methods for single-cell data analysis, including RNA-seq.  
  
## Citation  
  
## Contents 
- [AUCell](#AUCell) 
- [UCell](#UCell) 
= [irGSEA](irGSEA)
# List  
## packages  
  
### [AUCell] - [R] v1.22.0  
> [Aibar S, Aerts S (2016). “AUCell.”](https://doi.org/10.1038/nmeth.4463)  
> [Bioconductor](https://bioconductor.org/packages/release/bioc/html/AUCell.html)  
> [Vignettes](https://bioconductor.org/packages/release/bioc/vignettes/AUCell/inst/doc/AUCell.html)  
> AUCell allows to identify cells with active gene sets (e.g. signatures, gene modules...) in single-cell RNA-seq data.   
  
### [UCell] - [R] v1.3    
> [Massimo Andreatta & Santiago J Carmona (2021) CSBJ](https://doi.org/10.1016/j.csbj.2021.06.043)  
> [Bioconductor](https://bioconductor.org/packages/release/bioc/html/UCell.html)  
> [Vignettes](https://carmonalab.github.io/UCell_demo/UCell_Seurat_vignette.html)  
> Evaluating gene signatures in single-cell datasets based on the Mann-Whitney U statistic.  
   
### [irGSEA] - [R] v1.1.2    
> [Vignettes](https://chuiqin.github.io/irGSEA/)  
> Integrate all single cell rank-based gene set enrichment analysis and easy to visualize the results.  

### [ssGSEA] - [R] v2.0  
> [github](https://github.com/broadinstitute/ssGSEA2.0)  
> [web](https://www.genepattern.org/modules/docs/ssGSEAProjection/4#gsc.tab=0)  
> An extension of Gene Set Enrichment Analysis (GSEA), calculates separate enrichment scores for each pairing of a sample and gene set.  
  
### [singscore] - [R] v1.20.0  
> [github](https://github.com/DavisLaboratory/singscore)  
> [Reference](https://davislaboratory.github.io/singscore/articles/singscore.html)  
> [Vignettes](https://bioconductor.org/packages/release/bioc/vignettes/singscore/inst/doc/singscore.html)  
> Uses rank-based statistics to analyze each sample’s gene expression profile and scores the expression activities of gene sets at a single-sample level.  
