## SRA Toolkit 3.0.5 {:sra_toolkit}  
  
  \<terminal\>
[Github source](https://github.com/ncbi/sra-tools/wiki)  

## Trim Galore  0.6.5 {:trim_galore}

\<terminal, Bioconda\> 
[Github source](https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md)
[babraham bioinformat](https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/)

Trim Galore is a wrapper around Cutadapt and FastQC to consistently apply adapter and quality trimming to FastQ files, with extra functionality for RRBS data.

## FASTX-Toolkit {:fastq-tooklit}  
\<terminal\>  
[Github source](https://github.com/agordon/fastx_toolkit)  
  
The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.
Available Tools
+ **FASTQ-to-FASTA converter**:Convert FASTQ files to FASTA files.
+ **FASTQ Information**: Chart Quality Statistics and Nucleotide Distribution
+ **FASTQ/A Collapser**: Collapsing identical sequences in a FASTQ/A file into a single sequence (while maintaining reads counts)
+ **FASTQ/A Trimmer**: Shortening reads in a FASTQ or FASTQ files (removing barcodes or noise).
+ **FASTQ/A Renamer**: Renames the sequence identifiers in FASTQ/A file.
+ **FASTQ/A Clipper**: Removing sequencing adapters / linkers
+ **FASTQ/A Reverse-Complement**: Producing the Reverse-complement of each sequence in a FASTQ/FASTA file.
+ **FASTQ/A Barcode splitter**: Splitting a FASTQ/FASTA files containning multiple samples
+ **FASTA Formatte**r: changes the width of sequences line in a FASTA file
+ **FASTA Nucleotide Changer**: Convets FASTA sequences from/to RNA/DNA
+ **FASTQ Quality Filter**: Filters sequences based on quality
+ **FASTQ Quality Trimmer**: Trims (cuts) sequences based on quality
+ **FASTQ Masker**: Masks nucleotides with 'N' (or other character) based on quality

